src/Forms/GridFieldRefreshButton.php

```diff
@@ -164,8 +182,37 @@ public function hasActiveJob()
     public function handleRefresh()
     {
         if (!$this->hasActiveJob()) {
-            $injector = Injector::inst();
-            $injector->get(QueuedJobService::class)->queueJob($injector->create(CheckForUpdatesJob::class));
+            // Queue the job in the immediate queue
+            $job = Injector::inst()->create(CheckForUpdatesJob::class);
+            $jobDescriptorId = $this->getQueuedJobService()->queueJob($job, null, null, QueuedJob::IMMEDIATE);
+
+            // Check the job descriptor on the queue
+            $jobDescriptor = QueuedJobDescriptor::get()->filter('ID', $jobDescriptorId)->first();
+
+            // If the job is not immediate, change it to immediate and reschedule it to occur immediately
+            if ($jobDescriptor->JobType !== QueuedJob::IMMEDIATE) {
+                $jobDescriptor->JobType = QueuedJob::IMMEDIATE;
+                $jobDescriptor->StartAfter = null;
+                $jobDescriptor->write();
+            }
         }
     }
+
+    /**
+     * @return QueuedJobService
+     */
+    public function getQueuedJobService()
+    {
+        return $this->queuedJobService;
+    }
+
+    /**
+     * @param QueuedJobService $queuedJobService
+     * @return $this
+     */
+    public function setQueuedJobService($queuedJobService)
```

Comment:
Can you please type hint QueuedJobService

Reply:
👍